fairlearn/adversarial/_adversarial_mitigation.py

```diff
@@ -0,0 +1,822 @@
+
+    This model implements the supervised learning method proposed in
+    `"Mitigating Unwanted Biases with Adversarial Learning"
+    <https://dl.acm.org/doi/pdf/10.1145/3278721.3278779>`_ [1]_.
+    This algorithm takes as input two neural network
+    models, a predictor and an adversarial, defined either as a `PyTorch module
+    <https://pytorch.org/docs/stable/generated/torch.nn.Module.html>`_ or
+    `Tensorflow2 model
+    <https://www.tensorflow.org/api_docs/python/tf/keras/Model>`_. You train this
+    predictor using an API that is similar to estimators in `sklearn`.
+
+    As per [1]_, the neural network models :code:`predictor_model` and
+    :code:`adversary_model` may not have a discrete prediction at the end of the
+    model. Instead, even if we are dealing with discrete predictions, we may output
+    the sigmoidal or soft-max, but we may not output discrete integer predictions.
+
+    The distribution types of :code:`y` and :code:`sensitive_features`
+    are set by their preprocessor :code:`y_transform` and :code:`z_transform`
+    respectively. The default transformer :code:`FloatTransformer("auto")`
+    attempts to automatically infer
+    whether to assume binomial, multinomial, or normally distributed data.
+    You can force the transformer to assume one of the above by passing
+    :code:`FloatTransformer("binary")`, :code:`FloatTransformer("category")`,
+    :code:`FloatTransformer("continuous")` instead.
+    Loss functions and decision functions
+    for the target and sensitive features are also predefined for keywords
+    "binary", "category", and "continuous"
+    For more information, visit the user guide.
+
+    Parameters
+    ----------
+    backend : str, BackendEngine, default = 'auto'
+        The backend to use. Must be one of :code:`'torch'`, :code:`'tensorflow'`,
+        or :code:`'auto'` which indicates PyTorch, TensorFlow, or to
+        automatically infer
+        the backend from the :code:`predictor_model` and which are installed.
+        You can also pass in a BackendEngine class.
+
+    predictor_model : list, torch.nn.Module, tensorflow.keras.Model
+        The predictor model to train.
+        Instead of a neural network model, we can pass a list of keywords
+        :math:`k_1, k_2, \dots` that indicate either
+        the number of nodes :math:`k_i` (if :math:`k_i` is integer) or a keyword
+        that indicates an activation function (if :math:`k_i` is a string) or
+        a layer or activation function instance directly (if :math:`k_i` is
+        callable). The default parameter is :code:`[]`, which indicates
+        a neural network without any hidden layers.
+        However, the number of nodes in the input
+        and output layer are automatically inferred from data, and the final
+        activation function (such as softmax for categorical
+        predictors) are inferred from data.
+        If :code:`backend` is specified, we cannot pass a model
```

Comment:
This 'we cannot pass' does not match the 'You can also pass' in the previous parameter block.